src/org/zaproxy/zap/extension/ascanrulesAlpha/XssByCallbackScanner.java

```diff
@@ -0,0 +1,154 @@
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+
+public class XssByCallbackScanner extends AbstractAppParamPlugin implements ChallengeCallbackPlugin{
+
+    private static final Vulnerability VULN = Vulnerabilities.getVulnerability("wasc_8");
+    private static final int PLUGIN_ID = 40031;
```

Comment:
Needs to be registered https://github.com/zaproxy/zaproxy/blob/develop/src/doc/scanners.md

Reply:
yes, I'll make a PR

Reply:
PR created: https://github.com/zaproxy/zaproxy/pull/3940